site/src/components/station_dropdown_menu/filter_train.tsx

```diff
@@ -24,9 +24,10 @@ type Props = {
 export const FilterTrain = (props: Props) => {
   const { locale } = props
 
+  const [isReset, setIsReset] = React.useState(false)
   const { data: stations = [] } = useStations()
   const [selectedStation, setSelectedStation] =
-    React.useState<LocalizedStation | null>(null)
```

Comment:
Simplifying this to `undefined` instead of null so we can treat as an indicator that the form was reset. https://github.com/tailwindlabs/headlessui/pull/2660